src/docfx/restore/PublicTemplatePackage.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.Docs.Build
+{
+    internal class PublicTemplatePackage : Package
+    {
+        private readonly Uri _baseUrl;
+        private readonly FileResolver _fileResolver;
+
+        public PublicTemplatePackage(string baseUrl, FileResolver fileResolver)
+        {
+            _baseUrl = new Uri($"{baseUrl.TrimEnd('/')}/");
+            _fileResolver = fileResolver;
+        }
+
+        public override bool Exists(PathString path)
+        {
+            return _fileResolver.TryResolveFilePath(GetPath(path), out _);
+        }
+
+        public override IEnumerable<PathString> GetFiles()
+        {
+            throw new NotSupportedException();
+        }
+
+        public override Stream ReadStream(PathString path) => _fileResolver.ReadStream(GetPath(path));
+
+        public override PathString? TryGetPhysicalPath(PathString path)
+        {
+            if (_fileResolver.TryResolveFilePath(GetPath(path), out var fullPath))
```

Comment:
We should never hit this right?
```suggestion
            throw new NotSupportedException();
```